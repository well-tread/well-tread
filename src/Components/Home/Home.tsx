import React, { Component } from 'react';
import axios from 'axios';
// import TrailResults from './TrailResults/TrailResults';
import TrailResults from '../Search/TrailResults/TrailResults'
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import {
  withStyles,
  Theme,
  createMuiTheme,
  MuiThemeProvider
} from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#757575'
    },
    secondary: {
      main: '#FF5722'
    }
  }
});

export interface Props {}

export interface State {
  popularBikingTrails: any;
  popularHikingTrails: any;
  hikingArr: any;
  bikingArr: any;
  climbingArr: any;
  isResultsBack: boolean;
  latitude: number;
  longitude: number;
}

class Home extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      popularBikingTrails: [],
      popularHikingTrails: [],
      hikingArr: [],
      bikingArr: [],
      climbingArr: [],
      isResultsBack: false,
      latitude: 0,
      longitude: 0
    };
  }

  componentDidUpdate(prevProps: Props, prevState: State) {
    const { hikingArr, bikingArr, climbingArr, isResultsBack } = this.state;
    if (
      !isResultsBack &&
      (hikingArr.length > 0 || bikingArr.length > 0 || climbingArr.length > 0)
    ) {
      console.log('updating');
      this.setState({ isResultsBack: true });
    }
  }
  displayLocationInfo = (position: any) => {
    this.setState({
      latitude: position.coords.latitude,
      longitude: position.coords.longitude
    });
    const lng = position.coords.longitude;
    const lat = position.coords.latitude;
    axios
      .post('/trails/getPopularTrails', {
        lat: lat,
        lng: lng
      })
      //   .then(res => console.log('BIKE', res));
      .then(res => {
        const response = res.data;
        //console.log(response);
        this.setState({
          popularBikingTrails: response
        });
      });

    console.log(`longitude: ${lng} | latitude: ${lat}`);
  };
  componentDidMount() {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(this.displayLocationInfo);
    }
    // axios
    //   .post('/trails/getPopularTrails', {
    //     lat: this.state.latitude,
    //     lng: this.state.longitude
    //   })
    //   .then(res => console.log('BIKE', res));
    //   .then(res => {
    //     const response = res.data;
    //     console.log(response);
    //     this.setState({
    //       popularBikingTrails: response.trails
    //     });
    //   });

    // axios
    //   .get(
    //     `https://www.mtbproject.com/data/get-trails?lat=${
    //       this.state.latitude
    //     }&lon=${
    //       this.state.longitude
    //     }&maxDistance=200&minStars=4&maxResults=5&key=200430946-fc66551e94fef44057cb0cc88316bbec`
    //   )
    //   .then(res => {
    //     const response = res.data;
    //     console.log(res.data.trails);
    //     this.setState({
    //       popularBikingTrails: response.trails
    //     });
    //   })

    //   .catch(function(error) {
    //     console.log(error);
    //   });
    axios
      .get(
        `https://www.hikingproject.com/data/get-trails?lat=40.0274&lon=-105.2519&maxDistance=200&minStars=4&maxResults=5&key=200430946-fc66551e94fef44057cb0cc88316bbec`
      )
      .then(res => {
        const response = res.data;
        //console.log(res.data.trails);
        this.setState({
          popularHikingTrails: response.trails
        });
      })

      .catch(function(error) {
        console.log(error);
      });
  }

  render() {
    const {
      isResultsBack,
      hikingArr,
      bikingArr,
      climbingArr,
      popularBikingTrails,
      popularHikingTrails
    } = this.state;

    return (
      <MuiThemeProvider theme={theme}>
        <div>
          <div>
            <Typography variant='h5' color='primary' id='popular'>
              Top 5 Biking Trails Near You
            </Typography>
          </div>
          {this.state.popularBikingTrails.length > 0 ? <TrailResults
            // popularBikingTrails={popularBikingTrails}
            // popularHikingTrails={[]}
            hikingArr={[]}
            bikingArr={popularBikingTrails}
            climbingArr={[]}
          /> : <CircularProgress color='secondary'/>}
          <Typography variant='h5' color='primary' id='popular'>
            Top 5 Hiking Trails Near You
          </Typography>
          {this.state.popularHikingTrails.length > 0 ? <TrailResults
            // popularBikingTrails={[]}
            hikingArr={popularHikingTrails}
            // popularHikingTrails={popularHikingTrails}
            bikingArr={[]}
            climbingArr={[]}
          /> : <CircularProgress color='secondary'/>}
        </div>
      </MuiThemeProvider>
    );
  }
}

export default Home;
