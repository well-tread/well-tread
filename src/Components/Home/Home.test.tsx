import React from 'react';
import Home from './Home';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../ducks/store';

describe('snapshot', () => {
  test('Checking if Home component is rendering correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
