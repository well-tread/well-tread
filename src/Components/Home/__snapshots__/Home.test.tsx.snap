// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Checking if Home component is rendering correctly 1`] = `
<div>
  <div>
    <div>
      <h5
        className="MuiTypography-root-8 MuiTypography-h5-24 MuiTypography-colorPrimary-38 MuiTypography-gutterBottom-35 Weather-title-3"
      >
        city
         Weather Forecast
      </h5>
      <div
        className="Weather-cardContainer-6"
      >
        <div
          className="MuiPaper-root-45 MuiPaper-elevation1-48 MuiPaper-rounded-46 MuiCard-root-44 Weather-card-1"
        >
          <div
            className="MuiCardContent-root-72"
          >
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              Wednesday
            </p>
            <img
              className="Weather-icon-7"
              src=""
            />
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-deg-4"
            >
              0
              ยบ
            </p>
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              
            </p>
          </div>
        </div>
        <div
          className="MuiPaper-root-45 MuiPaper-elevation1-48 MuiPaper-rounded-46 MuiCard-root-44 Weather-card-1"
        >
          <div
            className="MuiCardContent-root-72"
          >
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              Wednesday
            </p>
            <img
              className="Weather-icon-7"
              src=""
            />
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-deg-4"
            >
              0
              ยบ
            </p>
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              
            </p>
          </div>
        </div>
        <div
          className="MuiPaper-root-45 MuiPaper-elevation1-48 MuiPaper-rounded-46 MuiCard-root-44 Weather-card-1"
        >
          <div
            className="MuiCardContent-root-72"
          >
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              Wednesday
            </p>
            <img
              className="Weather-icon-7"
              src=""
            />
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-deg-4"
            >
              0
              ยบ
            </p>
            <p
              className="MuiTypography-root-8 MuiTypography-body2-16 MuiTypography-colorPrimary-38 Weather-desc-5"
            >
              
            </p>
          </div>
        </div>
      </div>
    </div>
    <h5
      className="MuiTypography-root-8 MuiTypography-h5-24 MuiTypography-colorPrimary-38"
      id="popular"
    >
      Top 5 Biking Trails Near You
    </h5>
  </div>
  <div
    className="MuiCircularProgress-root-73 MuiCircularProgress-colorSecondary-77 MuiCircularProgress-indeterminate-75"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-78"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-79 MuiCircularProgress-circleIndeterminate-81"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <h5
    className="MuiTypography-root-8 MuiTypography-h5-24 MuiTypography-colorPrimary-38"
    id="popular"
  >
    Top 5 Hiking Trails Near You
  </h5>
  <div
    className="MuiCircularProgress-root-73 MuiCircularProgress-colorSecondary-77 MuiCircularProgress-indeterminate-75"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-78"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-79 MuiCircularProgress-circleIndeterminate-81"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <h5
    className="MuiTypography-root-8 MuiTypography-h5-24 MuiTypography-colorPrimary-38"
    id="popular"
  >
    Top 5 Running Trails Near You
  </h5>
  <div
    className="MuiCircularProgress-root-73 MuiCircularProgress-colorSecondary-77 MuiCircularProgress-indeterminate-75"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-78"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-79 MuiCircularProgress-circleIndeterminate-81"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
